format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -69,66 +69,68 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
         if (delegate.nextKeyValue()) {
           // skip to next if the current record is header
           if (skipHeader && delegate.getCurrentKey().get() == 0L) {
-            return delegate.nextKeyValue();
+            if (!delegate.nextKeyValue()) {
+              return false;
+            }
+          }
+          // getCurrentValue logic moved to nextKeyValue to avoid discrepancies in the data if there are multiple
```

Comment:
nit: Whoever is looking at this in the future won't know what was in getCurrentValue before. So let's change this comment to be something like `this logic must be in nextKeyValue to prevent multiple calls to getCurrentValue from advancing the delegate reader`

Reply:
Added